docs/setup-deploy-nixos.adoc

```diff
@@ -0,0 +1,71 @@
+= Deploy with NixOS
+
+NixOS deployment can be performed on the target machine itself or remotely
+by means of `nixos-rebuild --target-host`.
+
+These instructions will use `/etc/nixos` as the location for the configuration
+modules. This directory is only used at evaluation time. If you perform remote
+deployment, you will use some project directory in place of `/etc/nixos`.
+
+1.
+include::autogenerate/NixOSBootstrap.adoc[]
+
+2.
+include::autogenerate/NixOSImport.adoc[]
+
+3.
+Get a cluster join token.
+
+--
+- In the https://hercules-ci.com/dashboard[dashboard], find the account for which you would like to deploy the agent,
+- Click the "Agents" button and the button in "Generate token" tab. This produces a private token that should be protected like a password.
+- Copy the token into a plain text file `/var/lib/hercules-ci-agent/secrets/cluster-join-token.key` on the target machine.
+--
+
+4.
+Write a link:#binarycachespath[binary caches file], `binary-caches.json`. Continue with an applicable section below.
```

Comment:
AFAIK this is missing docs for setting up substituters.

Reply:
The instructions continue below, for what to do with the file.

When deploying locally, it can be read from the final location (evaluating as root)